File: CS/DXApplication5/CustomFunctions/IsWeekend.cs
using DevExpress.Data.Filtering;
using DevExpress.Utils.Svg;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DXSample {
    public class IsWeekendFunction : ICustomFunctionDisplayAttributes {
        public const string FunctionName = "IsWeekend";
        static readonly IsWeekendFunction instance = new IsWeekendFunction();

        public static void Register() {
            CriteriaOperator.RegisterCustomFunction(instance);
        }
        public static bool Unregister() {
            return CriteriaOperator.UnregisterCustomFunction(instance);
        }
        public FunctionCategory Category {
            get { return FunctionCategory.DateTime; }
        }
        public string Description {
            get { return "TEST TEST>"; }
        }

        public bool IsValidOperandCount(int count) {
            return count == 1;
        }
        public bool IsValidOperandType(int operandIndex, int operandCount, Type type) {
            return type == typeof(DateTime);
        }
        public int MaxOperandCount {
            get { return 1; }
        }
        public int MinOperandCount {
            get { return 1; }
        }
        public object Evaluate(params object[] operands) {
            DateTime dt = Convert.ToDateTime(operands[0]);
            return dt.DayOfWeek == DayOfWeek.Sunday || dt.DayOfWeek == DayOfWeek.Saturday;
        }
        public string Name {
            get { return FunctionName; }
        }
        public string DisplayName {
            get { return "Is weekend"; }
        }
        public object Image {
            get { return "DayView;Office2013"; }
        }
        public Type ResultType(params Type[] operands) {
            return typeof(bool);
        }
    }

    public class IsWeekendFunction2 : ICustomFunctionDisplayAttributes
    {
        string ICustomFunctionDisplayAttributes.DisplayName => throw new NotImplementedException();

        object ICustomFunctionDisplayAttributes.Image => throw new NotImplementedException();

        int ICustomFunctionOperatorBrowsable.MinOperandCount => throw new NotImplementedException();

        int ICustomFunctionOperatorBrowsable.MaxOperandCount => throw new NotImplementedException();

        string ICustomFunctionOperatorBrowsable.Description => throw new NotImplementedException();

        FunctionCategory ICustomFunctionOperatorBrowsable.Category => throw new NotImplementedException();

        string ICustomFunctionOperator.Name => throw new NotImplementedException();

        object ICustomFunctionOperator.Evaluate(params object[] operands)
        {
            throw new NotImplementedException();
        }

        bool ICustomFunctionOperatorBrowsable.IsValidOperandCount(int count)
        {
            throw new NotImplementedException();
        }

        bool ICustomFunctionOperatorBrowsable.IsValidOperandType(int operandIndex, int operandCount, Type type)
        {
            throw new NotImplementedException();
        }

        Type ICustomFunctionOperator.ResultType(params Type[] operands)
        {
            throw new NotImplementedException();
        }
    }
}

File: VB/DXApplication5/CustomFunctions/IsWeekend.vb
Imports DevExpress.Data.Filtering

Namespace DXSample
	Public Class IsWeekendFunction
		Implements ICustomFunctionDisplayAttributes

		Public Const FunctionName As String = "IsWeekend"
		Private Shared ReadOnly instance As New IsWeekendFunction()

		Public Shared Sub Register()
			CriteriaOperator.RegisterCustomFunction(instance)
		End Sub
		Public Shared Function Unregister() As Boolean
			Return CriteriaOperator.UnregisterCustomFunction(instance)
		End Function
        Public ReadOnly Property Category As FunctionCategory Implements ICustomFunctionOperatorBrowsable.Category
            Get
                Return FunctionCategory.DateTime
            End Get
        End Property
        Public ReadOnly Property Description As String Implements ICustomFunctionOperatorBrowsable.Description
            Get
                Return "TEST TEST>"
            End Get
        End Property

        Public Function IsValidOperandCount(count As Integer) As Boolean Implements ICustomFunctionOperatorBrowsable.IsValidOperandCount
            Return count = 1
        End Function
        Public Function IsValidOperandType(operandIndex As Integer, operandCount As Integer, type As Type) As Boolean Implements ICustomFunctionOperatorBrowsable.IsValidOperandType
            Return type Is GetType(Date)
        End Function
        Public ReadOnly Property MaxOperandCount As Integer Implements ICustomFunctionOperatorBrowsable.MaxOperandCount
            Get
                Return 1
            End Get
        End Property
        Public ReadOnly Property MinOperandCount As Integer Implements ICustomFunctionOperatorBrowsable.MinOperandCount
            Get
                Return 1
            End Get
        End Property
        Public Function Evaluate(ParamArray operands() As Object) As Object Implements ICustomFunctionOperator.Evaluate
            Dim dt As Date = Convert.ToDateTime(operands(0))
            Return dt.DayOfWeek = DayOfWeek.Sunday OrElse dt.DayOfWeek = DayOfWeek.Saturday
        End Function
        Public ReadOnly Property Name As String Implements ICustomFunctionOperator.Name
            Get
                Return FunctionName
            End Get
        End Property
        Public ReadOnly Property DisplayName As String Implements ICustomFunctionDisplayAttributes.DisplayName
            Get
                Return "Is weekend"
            End Get
        End Property
        Public ReadOnly Property Image As Object Implements ICustomFunctionDisplayAttributes.Image
            Get
                Return "DayView;Office2013"
            End Get
        End Property
        Public Function ResultType(ParamArray operands() As Type) As Type Implements ICustomFunctionOperator.ResultType
            Return GetType(Boolean)
        End Function
    End Class
End Namespace
